mathics/core/expression.py

```diff
@@ -743,7 +755,13 @@ def flatten(
         else:
             return self
 
-    def evaluate(self, evaluation) -> typing.Union["Expression", "Symbol"]:
+    def evaluate(self, evaluation) -> "BaseExpression":
+        """
+        Evaluate until reach a fixed point.
+        """
+        # comment @mmatera: Exceptions (like ReturnInterrupt) should be in a different module
```

Comment:
Yes, I agree. And at this stage it is important to me because we want to reevaluate the overall structure and starting off with something that is unnecessarily complicated doesn't hellp. 

So I have done this in  https://github.com/Mathics3/mathics-core/pull/134